consensus/istanbul/backend/announce.go

```diff
@@ -70,6 +72,102 @@ const (
 	LowFreqState
 )
 
+// AddressProvider provides the different addresses the announce manager needs
+type AddressProvider interface {
+	SelfNode() *enode.Node
+	ValidatorAddress() common.Address
+	IsValidating() bool
+}
+
+type ProxyContext interface {
+	GetProxiedValidatorEngine() proxy.ProxiedValidatorEngine
+}
+
+type AnnounceSupport interface {
+	// Gossip gossips protocol messages
+	Gossip(payload []byte, ethMsgCode uint64) error
+	// RetrieveValidatorConnSet returns the validator connection set
+	RetrieveValidatorConnSet() (map[common.Address]bool, error)
+	// Multicast will send the eth message (with the message's payload and msgCode field set to the params
+	// payload and ethMsgCode respectively) to the nodes with the signing address in the destAddresses param.
+	Multicast(destAddresses []common.Address, payload []byte, ethMsgCode uint64, sendToSelf bool) error
+}
+
+type AnnounceManagerConfig struct {
+	IsProxiedValidator bool
+	AWallets           *atomic.Value
```

Comment:
it's strange to have an atomic value in the `Config` object.

Reply:
Mmm true. I think this will get shuffled around when we have a clear understanding of what are all the dependencies and they can be regrouped.